resolver/map_test.go

```diff
@@ -174,7 +174,7 @@ func (s) TestAddressMap_Values(t *testing.T) {
 }
 
 func (s) TestEndpointMap_Length(t *testing.T) {
-	em := NewEndpointMap()
+	em := NewEndpointMap[any]()
```

Comment:
We can use `struct{}` as a more precise type parameter here, and `int` in the test cases below. Any reason to use `any`?

Reply:
Good idea.  I just did the simplest change possible for the tests, but using the proper type is better.